central/complianceoperator/v2/report/manager/results/results_aggregator.go

```diff
@@ -65,36 +67,96 @@ type aggregateResultsFn func(context.Context, string, *[]*report.ResultRow, *che
 func (g *Aggregator) GetReportData(req *report.Request) *report.Results {
 	resultsCSV := make(map[string][]*report.ResultRow)
 	reportResults := &report.Results{}
+	reportResults.ClustersData = make(map[string]*report.ClusterData)
 	for _, clusterID := range req.ClusterIDs {
-		clusterResults, clusterStatus, err := g.getReportDataForCluster(req.Ctx, req.ScanConfigID, clusterID, req.FailedClusters)
+		clusterData, err := g.getClusterData(req.Ctx, req.ScanConfigID, clusterID, req.ClusterData[clusterID])
+		if err != nil {
+			log.Errorf("Data unable to enhance cluster data for cluster %q: %v", clusterID, err)
+			continue
+		}
+		clusterResults, clusterStatus, err := g.getReportDataForCluster(req.Ctx, req.ScanConfigID, clusterID, clusterData)
 		if err != nil {
 			log.Errorf("Data not found for cluster %s", clusterID)
 			continue
 		}
+
 		resultsCSV[clusterID] = clusterResults
 		reportResults.TotalPass += clusterStatus.totalPass
 		reportResults.TotalFail += clusterStatus.totalFail
 		reportResults.TotalMixed += clusterStatus.totalMixed
+		reportResults.ClustersData[clusterID] = clusterData
 	}
 	reportResults.Clusters = len(req.ClusterIDs)
 	reportResults.Profiles = req.Profiles
 	reportResults.ResultCSVs = resultsCSV
 	return reportResults
 }
 
-func (g *Aggregator) getReportDataForCluster(ctx context.Context, scanConfigID, clusterID string, failedClusters map[string]*storage.ComplianceOperatorReportSnapshotV2_FailedCluster) ([]*report.ResultRow, *checkStatus, error) {
+func (g *Aggregator) getClusterData(ctx context.Context, scanConfigID, clusterID string, clusterData *report.ClusterData) (*report.ClusterData, error) {
+	if clusterData == nil {
+		return nil, errors.Errorf("cluster data for cluster %q is nil", clusterID)
+	}
+
+	scanNamesToProfileNames, err := g.scanDS.GetProfilesScanNamesByScanConfigAndCluster(ctx, scanConfigID, clusterID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to retrieve profiles associated with the ScanConfiguration %q in the cluster %q", scanConfigID, clusterID)
+	}
+	var profileNames []string
+	var scanNames []string
+	for scanName, profileName := range scanNamesToProfileNames {
+		profileNames = append(profileNames, profileName)
+		scanNames = append(scanNames, scanName)
+	}
+	clusterData.Profiles = profileNames
+	clusterData.Scans = scanNames
+	if clusterData.FailedInfo != nil {
+		clusterData.FailedInfo, err = g.getFailedClusterData(ctx, clusterData.FailedInfo, scanConfigID, clusterID)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to retrieve profiles associated the cluster %q", clusterData.ClusterId)
+		}
+	}
+	return clusterData, nil
+}
+
+func (g *Aggregator) getFailedClusterData(ctx context.Context, failedCluster *report.FailedCluster, scanConfigID, clusterID string) (*report.FailedCluster, error) {
```

Comment:
I got confused by the log saying `unable to retrieve profiles` when this function returns an error. 
It looks like this function takes the data from `.Scans`, queries the DB, and fills the data in `.Profiles`.
Wouldn't it make sense to implement it as part of `FailedCluster`?

For example:

```go
func (fc *FailedCluster) PopulateProfiles(scanDS whateverIsTheType) error {
    // do the logic
    fc.Profiles = profileNames
    return nil
}
```

or 

```go
func (fc *FailedCluster) PopulateProfiles(scanDS whateverIsTheType) ([]string, error) {
    // do the logic
    return profileNames, nil
}
```